src/modules/fancyzones/editor/FancyZonesEditor/Overlay.cs

```diff
@@ -116,19 +118,87 @@ public int CurrentDesktop
 
         private int _currentDesktop = 0;
 
-        public bool SpanZonesAcrossMonitors { get; set; }
+        public bool SpanZonesAcrossMonitors
+        {
+            get
+            {
+                return _spanZonesAcrossMonitors;
+            }
+
+            set
+            {
+                _spanZonesAcrossMonitors = value;
+
+                if (_spanZonesAcrossMonitors)
+                {
+                    Rect workArea = default(Rect);
+                    Rect bounds = default(Rect);
+
+                    foreach (Monitor monitor in Monitors)
+                    {
+                        workArea = Rect.Union(workArea, monitor.Device.WorkAreaRect);
+                        bounds = Rect.Union(bounds, monitor.Device.Bounds);
+                    }
+
+                    Monitors.Clear();
+                    Monitors.Add(new Monitor(bounds, workArea));
+                }
+            }
+        }
+
+        private bool _spanZonesAcrossMonitors;
 
         public Overlay()
         {
             UsedWorkAreas = new List<Rect>();
             Monitors = new List<Monitor>();
+
+            var monitors = MonitorInfoUtils.GetMonitors();
+            double primaryMonitorDPI = 96f;
+
+            for (int i = 0; i < monitors.Length; i++)
+            {
+                if (monitors[i].Primary)
+                {
+                    double monitorDPI;
+                    DpiAwareness.GetMonitorDpi(monitors[i].MonitorHandle, out monitorDPI, out monitorDPI);
+                    primaryMonitorDPI = monitorDPI;
+                    break;
+                }
+            }
+
+            for (int i = 0; i < monitors.Length; i++)
+            {
+                var monitor = monitors[i];
+
+                // get monitor dpi
+                double monitorDPI;
+                DpiAwareness.GetMonitorDpi(monitors[i].MonitorHandle, out monitorDPI, out monitorDPI);
```

Comment:
This code and the code at line 192 don't seem necessary, they were used in the test app but only for debugging reasons.